srml/support/src/double_map.rs

```diff
@@ -1,138 +0,0 @@
-// Copyright 2017-2019 Parity Technologies (UK) Ltd.
-// This file is part of Substrate.
-
-// Substrate is free software: you can redistribute it and/or modify
-// it under the terms of the GNU General Public License as published by
-// the Free Software Foundation, either version 3 of the License, or
-// (at your option) any later version.
-
-// Substrate is distributed in the hope that it will be useful,
-// but WITHOUT ANY WARRANTY; without even the implied warranty of
-// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-// GNU General Public License for more details.
-
-// You should have received a copy of the GNU General Public License
-// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
-
-//! An implementation of double map backed by storage.
-
-use crate::rstd::prelude::*;
-use crate::codec::{Codec, Encode};
-use crate::storage::unhashed;
-use sr_std::borrow::Borrow;
-
-/// An implementation of a map with a two keys.
-///
-/// It provides an important ability to efficiently remove all entries
-/// that have a common first key.
-///
-/// # Mapping of keys to a storage path
-///
-/// The storage key (i.e. the key under which the `Value` will be stored) is created from two parts.
-/// The first part is a hash of a concatenation of the `PREFIX` and `Key1`. And the second part
-/// is a hash of a `Key2`.
-pub trait StorageDoubleMap {
```

Comment:
As I've mentioned [in my comment](https://github.com/paritytech/substrate/pull/2002#discussion_r266493543), I don't see a reason for deleting `StorageDoubleMap`, it is just contracts don't use it anymore. In fact, there is [a PR](https://github.com/paritytech/substrate/pull/1918) that promotes `StorageDoubleMap` to be a first-class citizen in decl_storage. 

Or am I missing something?

Reply:
My fault, I understood your comment as a request for removal, personally I wanted to keep it but considering it was not used I did the removal. (you probably wanted to remove the TODO, not the StorageDoubleMap :)
Seeing that it will be move by the cited PR, I can restore it (good thing I kept the commit hash) and also remove the TODO comment.

Reply:
To be clear: I wanted to keep `StorageDoubleMap`, remove `StorageOf` (which implements `StorageDoubleMap`) because it is not needed, is it?

Reply:
Not needed, should be restored at this point, I am not removing StorageOf in the PR as it will be redundant with #1918, thinking twice it will not conflict if I also remove it here: I will remove it in next commit.